### DISCUSSION TITLE
How to organize (auto-closed) issues solved on master branch but not in a release yet?

### DISCUSSION THREAD

**Discussion by dtugend** [community/community#21808] [New to GitHub]
<p><strong>Situation:</strong></p>
<ul>
<li>The repository has a master branch for code contributions / pull-request.</li>
<li>The issues section is also used by “normal” users (e.g. for questions / or to see yet unresolved bugs (e.g. partially with workarounds)).</li>
</ul>
<p><strong>Problem:</strong></p>
<ul>
<li>Closing an issue for the master branch (e.g. from a commit), also closes the issue in the issues section, even though there’s no release with that commit yet, so people will get confused sooner or later, since it’s marked as closed.</li>
</ul>
<p><strong>How would or already do you go about organizing this / or solving this in another way?</strong></p>
<hr>
<p><strong>Things we already tried:</strong></p>
<p>In the past we had a special upcoming branch, where I pushed changes to, so after a new version was released, I would only then merge it to master, thus auto-closing the issues.</p>
<p>The problem with that is, that coders usually want to make pull-requests onto the master branch.</p>
<hr>
<p><strong>Suggestion (Maybe not doable with reasonable effort?):</strong></p>
<p>Ideally I wouldn’t want issues auto-closed until the commit is merged into latest release (not pre-release).</p>

**Comment by nyahbhinghiPrincess** [comment, marked as answer]
<p>Hello <a class="mention" href="/dtugend">@dtugend</a>!</p>
<p>Thank you for sharing your use case!</p>
<p>Some developers have taken the approach of <a href="https://github.community/t/create-releases-for-old-commits/872">creating a release from an older commit</a>, <em>although, that approach does not address the sight of a closed issue for your users</em>.  Aside from using a separate repository just to collect issues from the external users, some have taken to editing the commit message to avoid automatic closure of the mentioned issue.</p>
<p>Our community members may have more suggestions and creative approaches to share on this. </p>
<p>In the meantime, I have also taken your suggestion and passed it along to the appropriate teams!  I don’t have an ETA on when or if this request will be implemented but this is in good hands.</p>

**Comment by lee-dohm** [comment]
<p>For <a href="https://atom.io" rel="nofollow noopener">the Atom editor</a>, we have <a href="https://github.community/t/how-to-accept-selective-commits-for-one-branch/1120/4">a fairly complex release process that I’ve documented previously</a>. The tl;dr is there are three main branches:</p>
<ul>
<li>
<code>master</code> where all new changes go</li>
<li>A beta branch, created from <code>master</code> when a new beta version is created</li>
<li>A stable branch, created from the beta branch when a new stable version is created</li>
</ul>
<p>We close issues as soon as they are fixed on <code>master</code>. If people ask when it will make it to the stable release, we tell them the version number associated with <code>master</code> and say that it is scheduled to be available in that release (<a href="https://github.com/atom/atom/issues/17940#issuecomment-416736314" rel="nofollow noopener">example</a>).</p>
<p>Yes, there is some overhead to this. But we’ve found that it balances out the problems of:</p>
<ol>
<li>Keeping the development process streamlined for the maintainers and contributors</li>
<li>Making the triage process easy for the maintainers</li>
<li>New GitHub users can be taught how to figure out the information for themselves (in other words, if the current stable version is v1.23.0 and the beta is v1.24.0-beta0, then something that was merged recently is going to be available when v1.25.0 reaches stable)</li>
</ol>
<p>And since Atom now has nightly releases, we can always tell them if it is really important they have the bug fix right away, they can volunteer to test new releases (that might have other bugs in them) by installing the nightly version. Most of them opt to wait 😀</p>
<p>I hope that helps!</p>

**Comment by dtugend** [comment]
<p>Thank you both for your replies and ideas!</p>
<p>I’ll probably try to assign the closed issues in master branch to a milestone called “upcoming” and when there’s a new release merging from master branch, I will rename the milestone to that release tag and then close it, so people can see if it’s already released and in which version or at least being pointed at that information. It’s a bit more work, but I guess it will work.</p>

**Comment by sean-mcmanus** [comment]
<p>How does editing a commit message prevent auto-close of a linked issue?</p>
<p>I would prefer there be an additional “Resolved” state instead of just Open/Closed, so the issues that have been fixed but not released can be easily seen as Resolved and then all the Resolved issues can be automatically moved to the Closed state after the linked milestone is closed.</p>
<p>Up to now, we’ve just been keeping issues Open until the fix has been shipped, but then some of those issues get closed via linked PRs.</p>
<p>EDIT: Another option would be to add several detailed resolution categories like <a href="https://developercommunity.visualstudio.com" target="_blank" rel="nofollow noopener">https://developercommunity.visualstudio.com</a> does, like “Fixed-  Pending Release”, “Won’t Fix”, “Closed - Duplicate”, “Closed - Fixed”, “Closed - Lower Priority”, “Closed - Not a Bug”, “Under Investigation”. We sort of do that via our “Fixed Pending Release” label, but our issues have like 5 labels sometimes and it would be better if the “resolution” was separated out somehow as being a special label.</p>

**Comment by dtugend** [comment]
<p>That’s actually a very good idea that I would like too as solution, if it is possible.<br>
The commit message keywords should also reflect that feature if possible in my opinion.</p>
<p>Have you already opened a feature request related to that: <a href="https://support.github.com/contact/feedback" rel="nofollow noopener">https://support.github.com/contact/feedback</a> - otherwise I will do?</p>
<p>(Be aware that they will take quite some time to reply to the feature request, since they have quite some support volumne at the moment it seems.)<br>
<strong>Edit: I already submitted it now.</strong></p>
